plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/DeployMetaDataParameterValue.java

```diff
@@ -0,0 +1,69 @@
+package com.epam.grandhackathon.deployment.sphere.plugin;
+
+import hudson.EnvVars;
+import hudson.model.ParameterValue;
+import hudson.model.AbstractBuild;
+import hudson.util.VariableResolver;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.export.Exported;
+
+import com.epam.grandhackathon.deployment.sphere.plugin.metadata.Constants;
+
+@EqualsAndHashCode(callSuper = false)
+@ToString(callSuper = false)
+public class DeployMetaDataParameterValue extends ParameterValue {
+  @Exported(visibility=4)
+  private String environmentKey;
+  private String buildVersion;
+  private String applicationName;
+
+  @DataBoundConstructor
+  public DeployMetaDataParameterValue(String name, String environmentKey, String buildVersion, String applicationName) {
+    this(name, environmentKey, buildVersion, applicationName, null);
+  }
+
+  public DeployMetaDataParameterValue(String name, String environmentKey, String buildVersion, String applicationName, String description) {
+    super(name, description);
+    this.environmentKey = environmentKey;
+    this.buildVersion = buildVersion;
+    this.applicationName = applicationName;
+  }
+
+  @Exported
+  public String getEnvironmentKey(){
+	  return environmentKey;
+  }
+  
+  @Exported
+  public String getBuildVersion(){
+	  return buildVersion;
+  }
+  
+  @Exported
+  public String getApplicationName(){
+	  return applicationName;
+  }
+  
+  @Override
+  public void buildEnvVars(AbstractBuild<?,?> build, EnvVars env) {
+    env.put(Constants.ENV_NAME, environmentKey);
+    env.put(Constants.BUILD_VERSION, buildVersion);
+    env.put(Constants.BUILD_APP_NAME, applicationName);
+  }
+
+  @Override
+  public VariableResolver<String> createVariableResolver(AbstractBuild<?, ?> build) {
+    return new VariableResolver<String>() {
+      public String resolve(String name) {
+        return Constants.ENV_NAME.equals(name) ? 
```

Comment:
a bit complex code, please make it simpler by extracting into methods with proper names